plasmapy/particles/particle_class.py

```diff
@@ -96,6 +97,56 @@ def _category_errmsg(particle, category: str) -> str:
     return errmsg
 
 
+class ParticleJSONDecoder(json.JSONDecoder):
+
+    def __init__(self, *args, **kwargs):
+        json.JSONDecoder.__init__(self, object_hook=self.object_hook, *args, **kwargs)
+
+    def object_hook(self, json_dict):
+        """
+        Returns the appropriate particle object corresponding to the input JSON
+        string. If not a particle, returns a dictionary object. Logic Implementation for
+        plasmapy.particles.from_json method.
+
+        Examples
+        --------
+        >>> import plasmapy
+        >>> input_str = '{"plasmapy_particle": {"type": "Particle", "module": "plasmapy.particles.particle_class", "date_created": "2020-06-08", "init_args": {"argument": "Pb"}}}'
```

Comment:
Including `date` is an interesting idea. This is the exact kind of safeguard against future functionality changes that I would forget if writing this myself.

Are you using the `module` anywhere?

Reply:
> Including date is an interesting idea. This is the exact kind of safeguard against future functionality changes that I would forget if writing this myself.

Credits to @rocco8773 for the suggestion. :)

> Are you using the module anywhere?

Not for right now. The `module` stores where the class was defined and is useful for tracking in future if particles do stem from other source code files. It doesn't play a role in the deserialization. Another interesting suggestion from @rocco8773 :)

For reference, the suggestions can be found here: https://github.com/PlasmaPy/PlasmaPy/pull/833#discussion_r436424242

Reply:
It gets so hard to follow discussions in previous reviews sometimes... :(

Reply:
Indeed it does. Unfortunately, I made the problem worse by splitting the PRs. Oops :(

Reply:
Nah, no worries. My bad, honestly :)